oracle/scripts/initialChecks.js

```diff
@@ -1,15 +1,33 @@
 require('../env')
 const Web3 = require('web3')
-const { ERC677_BRIDGE_TOKEN_ABI, getTokenType } = require('../../commons')
+const {
+  ERC677_BRIDGE_TOKEN_ABI,
+  FOREIGN_ERC_TO_ERC_ABI,
+  FOREIGN_ERC_TO_NATIVE_ABI,
+  FOREIGN_NATIVE_TO_ERC_ABI,
+  getTokenType
+} = require('../../commons')
 
 async function initialChecks() {
-  const { ERC20_TOKEN_ADDRESS, ORACLE_BRIDGE_MODE, COMMON_FOREIGN_RPC_URL, COMMON_FOREIGN_BRIDGE_ADDRESS } = process.env
+  const { ORACLE_BRIDGE_MODE, COMMON_FOREIGN_RPC_URL, COMMON_FOREIGN_BRIDGE_ADDRESS } = process.env
   const result = {}
+  const foreignWeb3 = new Web3(new Web3.providers.HttpProvider(COMMON_FOREIGN_RPC_URL))
 
   if (ORACLE_BRIDGE_MODE === 'ERC_TO_ERC') {
-    const foreignWeb3 = new Web3(new Web3.providers.HttpProvider(COMMON_FOREIGN_RPC_URL))
-    const bridgeTokenContract = new foreignWeb3.eth.Contract(ERC677_BRIDGE_TOKEN_ABI, ERC20_TOKEN_ADDRESS)
+    const bridge = new foreignWeb3.eth.Contract(FOREIGN_ERC_TO_ERC_ABI, COMMON_FOREIGN_BRIDGE_ADDRESS)
+    result.ERC20_TOKEN_ADDRESS = await bridge.methods.erc20token().call()
+  } else if (ORACLE_BRIDGE_MODE === 'ERC_TO_NATIVE') {
+    const bridge = new foreignWeb3.eth.Contract(FOREIGN_ERC_TO_NATIVE_ABI, COMMON_FOREIGN_BRIDGE_ADDRESS)
+    result.ERC20_TOKEN_ADDRESS = await bridge.methods.erc20token().call()
+  } else if (ORACLE_BRIDGE_MODE === 'NATIVE_TO_ERC') {
+    const bridge = new foreignWeb3.eth.Contract(FOREIGN_NATIVE_TO_ERC_ABI, COMMON_FOREIGN_BRIDGE_ADDRESS)
+    result.ERC20_TOKEN_ADDRESS = await bridge.methods.erc677token().call()
```

Comment:
From my perspective `ERC20_TOKEN_ADDRESS` should not be used in the `native-to-erc` mode.

Reply:
Updated, checking the token address only for `ERC_TO_ERC` and `ERC_TO_NATIVE`